tsconfig.base.json

```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
-    "module": "commonjs",
-    "moduleResolution": "nodenext",
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext",
```

Comment:
My 2c: it'll be less painful in future if we switch to ESM output (which we'd get with `module: nodenext` if we have `type: module` in each package.json).

Also `moduleResolution: bundler` would be more correct since we use esbuild, except for maybe whatever the test harness consumes, since we don't disable tsc emit?

Reply:
Yeah we should prob switch to bundler. Forgot bout that option

Reply:
So if we switch to bundler, what should we use for `module`? Should we revert it, or use `NodeNext` anyway?

Reply:
Having the package.json dictate whether the emit is CJS or ESM sounds reasonable.

I'm not sure continuing to emit CommonJS is a great idea though given that locks us out of using ESM-only libraries.

Reply:
Looks like we [should be using ESNext with bundler](https://www.typescriptlang.org/docs/handbook/modules/reference.html#implied-and-enforced-options-2)

Reply:
Ok I eventually gave up on trying to get `type: "module"` to work. We are using `webpack.config.ts`, and importing from other packages in our monorepo. That caused all sorts of craziness becasue `webpack.config.ts` needs to be a commonjs module, so the fact that everything else had `package.json` saying it is es module caused it to lose its mind. Feel free to look through the CI failures as I flailed around in the commit history

If there's no obvious fix, I'd be inclined to ship what I have now

Reply:
...and looks like module resolution bundler isn't working either; something to do how we launch our vscode tests. I would be inclined to revert to the last passing commit (63174732e525e090fa4a7f8540fc8e347ed84df6) and ship, as I've already spent the better part of a day on this 😅

Reply:
@auscompgeek see above—unless you have some silver bullet to get `type: "module"` or `moduleResolution: "bundler"` to work, I'd be inclined to roll back to the last passing commit and tackle other stuff later. Would be nice to get TS 5.2 features and I don't think that last working commit is a step backward otherwise